src/coprocessor/dag/rpn_expr/impl_time.rs

```diff
@@ -0,0 +1,166 @@
+                "Jan January 01 1 1st 01 1 001 0 12 00 AM 12:00:00 AM 00:00:00 00 123456 52
+                4294967295 0000 00 %",
+            ),
+            (
+                "2016-09-3 00:59:59.123456",
+                "abc%b %M %m %c %D %d %e %j %k %h %i %p %r %T %s %f %U
+                %u %V %v %a %W %w %X %x %Y %y!123 %%xyz %z",
+                "abcSep September 09 9 3rd 03 3 247 0 12 59 AM 12:59:59 AM 00:59:59 59 123456 35
+                35 35 35 Sat Saturday 6 2016 2016 2016 16!123 %xyz z",
+            ),
+            (
+                "2012-10-01 00:00:00",
+                "%b %M %m %c %D %d %e %j %k %H %i %p %r %T %s %f %v
+                %x %Y %y %%",
+                "Oct October 10 10 1st 01 1 275 0 00 00 AM 12:00:00 AM 00:00:00 00 000000 40
+                2012 2012 12 %",
+            ),
+        ];
+        for (date, format, expect) in cases {
+            let date = Some(DateTime::parse_utc_datetime(date, 6).unwrap());
+            let format = Some(format.as_bytes().to_vec());
+            let expect = Some(expect.as_bytes().to_vec());
+
+            let output = RpnFnScalarEvaluator::new()
+                .push_param(date.clone())
+                .push_param(format.clone())
+                .evaluate(ScalarFuncSig::DateFormatSig)
+                .unwrap();
+            assert_eq!(output, expect, "{:?} {:?}", date, format);
+        }
+
+        //                // TODO: pass this test after refactoring the issue #3953 is fixed.
+        //                {
+        //                    let format: Option<Bytes> =  Some("abc%b %M %m %c %D %d %e %j".as_bytes().to_vec());
+        //                    let time: Option<DateTime> = Some( DateTime::parse_utc_datetime("0000-00-00 00:00:00", 6).unwrap());
+        //
+        //                    let mut cfg = EvalConfig::new();
+        //                    cfg.set_flag(Flag::IN_UPDATE_OR_DELETE_STMT)
+        //                        .set_sql_mode(SqlMode::NO_ZERO_DATE | SqlMode::STRICT_ALL_TABLES);
+        //                    let ctx = EvalContext::new(Arc::new(cfg));
+        //
+        //                    let output = RpnFnScalarEvaluator::new()
+        //                        .context(ctx)
+        //                        .push_param(time.clone())
+        //                        .push_param(format)
+        //                        .evaluate::<Bytes>(ScalarFuncSig::DateFormatSig);
+        //                    assert!(output.is_err());
+        //                }
+
+        {
+            let format: Option<Bytes> = None;
+            let time: Option<DateTime> = None;
+
+            let mut cfg = EvalConfig::new();
+            cfg.set_flag(Flag::IN_UPDATE_OR_DELETE_STMT)
+                .set_sql_mode(SqlMode::NO_ZERO_DATE | SqlMode::STRICT_ALL_TABLES);
+            let ctx = EvalContext::new(Arc::new(cfg));
+
+            let output = RpnFnScalarEvaluator::new()
+                .context(ctx)
+                .push_param(time.clone())
```

Comment:
Why not simply `.push_param(None)`

Reply:
`Option::<X>::None` could be written as `None::<X>` instead.